store/src/java/com/zimbra/cs/account/JWTCache.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2017 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.account;
+
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.zimbra.common.localconfig.LC;
+import com.zimbra.common.util.MapUtil;
+
+public class JWTCache {
+
+    private static final Map<String, ZimbraJWT> JWT_CACHE = MapUtil.newLruMap(LC.zimbra_authtoken_cache_size.intValue());
```

Comment:
Could you use CacheBuilder - see https://github.com/google/guava/wiki/CachesExplained or even ConcurrentHashMap?  Suspect they have better concurrency handling (less locking)

Reply:
used the guava library cache.